buoy_gazebo/src/PolytropicPneumaticSpring/PolytropicPneumaticSpring.cpp

```diff
@@ -123,6 +127,81 @@ double SdfParamDouble(
   return _sdf->Get<double>(_field, _default).first;
 }
 
+void PolytropicPneumaticSpring::manageCommandTimer(SpringState & state)
+{
+  if (state.valve_command || state.pump_command) {
+    if (!state.command_watch.Running()) {
+      if (state.valve_command) {
+        ignmsg << "Valve opening (" << \
+          std::chrono::duration_cast<std::chrono::seconds>(state.command_duration).count() << \
+          "s)" << std::endl;
+      }
+      if (state.pump_command) {
+        ignmsg << "Pump on (" << \
+          std::chrono::duration_cast<std::chrono::seconds>(state.command_duration).count() << \
+          "s)" << std::endl;
+      }
+      state.command_watch.Start(true);
+    } else {
+      if (state.command_watch.ElapsedRunTime() >= \
+        state.command_duration)
+      {
+        state.command_watch.Stop();
+
+        if (state.valve_command) {
+          state.valve_command = false;
+          ignmsg << "Valve closed" << std::endl;
+        }
+
+        if (state.pump_command) {
+          state.pump_command = false;
+          ignmsg << "Pump off" << std::endl;
+        }
+      }
+    }
+  } else if (!state.valve_command && !state.pump_command) {
```

Comment:
I think this may be the same as

```suggestion
  } else {
```

Just checking if that was the intention. I can see an argument for being explicit, but for someone trying to follow the logic it may add some complexity.

Reply:
I was being explicit (zen of python? :stuck_out_tongue_winking_eye: ) but I actually just made a pass at cleaning up logic and this went away.